pkgs/flox-activation-scripts/default.nix

```diff
@@ -38,11 +42,14 @@ let
     # Note that substitution doesn't work with variables containing "-"
     # so we need to create and use alternative names.
     process_compose = process-compose;
+    # If the flox-activations package is available, use it,
+    # otherwise copy the binary from the environment into the store,
+    # so that sandboxed builds and flox built containers can access it.
     flox_activations =
       if flox-activations != null then
         "${flox-activations}/bin/flox-activations"
       else
-        "$FLOX_ACTIVATIONS_BIN";
+        "${builtins.path { path = builtins.getEnv "FLOX_ACTIVATIONS_BIN"; }}";
```

Comment:
blocking: does that break this dependency being dynamic?

Reply:
you'll have to `just build` or `just build-activation-scripts` -- the latter is enough at a minimum.
This package is used in sandboxes (both nix and docker), i dont think we can get to this without going through the store.

Reply:
Can we just do a nix build if we need to test in a container?

Reply:
> Can we just do a nix build if we need to test in a container?

I dont exactly understand what you mean.


Because flox-activation-scripts get embedded in the container, you cant _run_ the container itself it will err with something like:


```
# when resolving the path at eval time
/.../flox-activations: No such file or directory

# when resolving the variable at runtime
--runtime-dir: No such file or directory
```


we can opt to override FLOX_BUILDENV_NIX for the nix built version in tests, but we'd also need to do so in build unit tests that make use of the nix sandbox (at least on linux).

Besides tests this would also be necessary to run containers and `flox build` even outside of tests in development which.

Reply:
I meant for getting things working in a container, doing a `nix build .#flox-activation-scripts` rather than `.#developmentPackages`. But I hadn't thought about all those cases in testing. It not being dynamic still isn't my favorite but I don't have a better suggestion at the moment so LGTM